src/components/skills-quiz/SkillsContextProvider.jsx

```diff
@@ -0,0 +1,37 @@
+import React, { createContext, useReducer } from 'react';
+import PropTypes from 'prop-types';
+import {
+  SET_KEY_VALUE,
+} from './data/constants';
+
+export const SkillsContext = createContext();
+
+const reducer = (state, action) => {
+  const nextState = { ...state };
+  switch (action.type) {
+    case SET_KEY_VALUE:
+      delete nextState.page;
+      nextState[action.key] = action.value;
+      return nextState;
+  default:
+    return state;
+  }
+};
+
+export function SkillsContextProvider({ children, initialState = { goal: 'Goal' } }) {
```

Comment:
Question: Shouldn't we set an initial state for all the things that we plan to store in this Context? like for example `interestedJobs`?

Reply:
Yes, we need to set all those values in the initial states which are needed at the first render. In the case of this `context`, only `goal` is needed and the rest of the values will be used only when they exist. i.e `insterstedJobs` are rendered only when a `job` is selected in the `dropdown`.Hence no need to define it in `initialstate`.

Reply:
Okay, sounds good!